### PR TITLE
Modified api to include /hosts and /paths resources. API aligned with…

### PR DESCRIPTION
**Pull request by c00ler**
… the implementation.

### CODE REVIEW

**Comment by danpersa** [comment]
lgtm

**Comment by danpersa** [comment]
let's not merge it to master, but create a new feature branch :)

**Comment by c00ler** [comment]
@grassator Please take a look at the changes. Ignore last comment from @danpersa.

File: swagger.yaml [hunk 1]
```diff
@@ -412,8 +509,65 @@ definitions:
...
+      name:
+        type: string
+    example: |
+      {
+        "id": "1",
+        "name": "zalando.de"
+      }
+  PathIn:
+    required:
+      - uri
```
Comment (grassator) [inline swagger.yaml]: Is empty URI invalid? If so I think we need to explicitly mark it in the swagger

File: swagger.yaml [hunk 2]
```diff
@@ -412,8 +509,65 @@ definitions:
...
+          type: string
+    example: |
+      {
+        "uri": "/cart",
+        "host_ids": ["1", "2"]
+      }
+  PathOut:
+    required:
+      - id
+      - team
```
Comment (grassator) [inline swagger.yaml]: The Route has  `owned_by_team` and `created_by`, why does this have only one field and named differently?

File: swagger.yaml [hunk 3]
```diff
@@ -366,7 +460,8 @@ definitions:
       value:
         type: string
         description: |
-          Depending on the `type`, this is either a plain string for `STRICT` or ECMA-262 valid regular expression for `REGEXP`.
+          Depending on the `type`, this is either a plain string for `STRICT` or ECMA-262 valid 
+          regular expression for `REGEXP`.
   Matcher:
     properties:
       host_matcher:
```
Comment (grassator) [inline swagger.yaml]: Didn't we discuss dropping the matchers in favor of predicates?

File: swagger.yaml (hunk 1 above)
Comment (c00ler) [inline swagger.yaml]: Yes, empty URI should be prohibited. How can I do it in swagger spec? Just a description?

File: swagger.yaml (hunk 2 above)
Comment (c00ler) [inline swagger.yaml]: Good point. I also added created_at.

File: swagger.yaml [hunk 4]
```diff
@@ -28,40 +28,106 @@ securityDefinitions:
...
+        - company:
+          - routes.read
+      summary: A list of available paths.
+      parameters:
+        - name: team
           in: query
-          description: The name of the route to filter for
+          type: string
           required: false
+        - name: uri
```
Comment (grassator) [inline swagger.yaml]: What about filtering by hosts?

File: swagger.yaml [hunk 5]
```diff
@@ -302,118 +387,127 @@ definitions:
...
         description: |
-          Defines the type of matching that will be used for a path.
-          The usage of REGEXP path is only allowed for users with the routes.write_regex scope
-      match:
+          The name of the predicate, as it is definined in Skipper
+      args:
+        description: The arguments for the predicate, can be string or float64
+        type: array
+        items:
+          type: string
```
Comment (grassator) [inline swagger.yaml]: Why isn't float64 mentioned here?

File: swagger.yaml (hunk 1 above)
Comment (grassator) [inline swagger.yaml]: regexp `.+`?

File: swagger.yaml (hunk 4 above)
Comment (c00ler) [inline swagger.yaml]: This is not the final version. Let's add it later, when we will have requests for this feature.

File: swagger.yaml (hunk 1 above)
Comment (c00ler) [inline swagger.yaml]: done

**Pull request merged**